File: python/spear/services/world_registry_service.py
#
# Copyright (c) 2025 The SPEAR Development Team. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
# Copyright (c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
#

import spear

class WorldRegistryService(spear.Service):
    def __init__(self, entry_point_caller):

        # do this after initializing local state
        super().__init__(entry_point_caller=entry_point_caller)

    def get_world_descs(self):
        return self.entry_point_caller.call_on_worker_thread("get_world_descs", None)

    def remove_world(self, world):
        return self.entry_point_caller.call_on_game_thread("remove_world", None, world)

File: cpp/unreal_plugins/SpServices/Source/SpServices/WorldRegistryService.cpp
//
// Copyright (c) 2025 The SPEAR Development Team. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
// Copyright (c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
//

#include "SpServices/WorldRegistryService.h"

#include <memory> // std::make_unique, std::unique_ptr

#include "SpServices/EngineService.h"
#include "SpServices/EntryPointBinder.h"
#include "SpServices/RpcServer.h"

template <CUnrealEntryPointBinder TEntryPointBinder>
std::unique_ptr<WorldRegistryService> WorldRegistryService::create(TEntryPointBinder* unreal_entry_point_binder) { return std::make_unique<WorldRegistryService>(unreal_entry_point_binder); }

template std::unique_ptr<WorldRegistryService> WorldRegistryService::create<EngineService<RpcServer>>(EngineService<RpcServer>*);
